### ISSUE TITLE
Elastic Search / Kibana 6.x compatibility

### ISSUE THREAD

**Issue by daboom** [stagemonitor/stagemonitor#345]
Just to know if it is compatible with the new Elastic Search 6.x

**Comment by felixbarny** [comment]
Did not try yet ;)
But if you want it would be great if you could try out and give feedback

**Comment by daboom** [comment]
Tried with elasticsearch-6.0.0-rc2 on Windows.
At loading time, it creates the indexes but it looks like it'is not able to do some bulks on the .kibana index

<code>
15:21:59.223 [async-elasticsearch] WARN org.stagemonitor.core.elasticsearch.ElasticsearchClient$BulkErrorReportingResponseHandler - Error(s) while sending a _bulk request to elasticsearch:
 - Rejecting mapping update to [.kibana] as the final mapping would have more than 1 type: [doc, dashboard]
 - Rejecting mapping update to [.kibana] as the final mapping would have more than 1 type: [visualization, doc]
</code>
 
Here is Elastic Search log:

<code>
[2017-11-08T15:21:59,220][DEBUG][o.e.a.b.TransportShardBulkAction] [.kibana][0] failed to execute bulk item (index) BulkShardRequest [[.kibana][0]] containing [16] requests
java.lang.IllegalArgumentException: Rejecting mapping update to [.kibana] as the final mapping would have more than 1 type: [visualization, doc]
        at org.elasticsearch.index.mapper.MapperService.internalMerge(MapperService.java:494) ~[elasticsearch-6.0.0-rc2.jar:6.0.0-rc2]
        at org.elasticsearch.index.mapper.MapperService.internalMerge(MapperService.java:350) ~[elasticsearch-6.0.0-rc2.jar:6.0.0-rc2]
        at org.elasticsearch.index.mapper.MapperService.merge(MapperService.java:282) ~[elasticsearch-6.0.0-rc2.jar:6.0.0-rc2]
</code>

**Comment by daboom** [comment]
Is there a plan on this?

**Comment by felixbarny** [comment]
The plan is to support elastic 6, but there is no set schedule. Do you think you could contribute the support for elastic 6?

**Comment by tylersmalley** [comment]
You should not be inserting documents into the Kibana index. Doing so also breaks users who have upgraded to 5.6 and have re-indexed to prepare for the 6.0 migration. What happens is similar to the situation outlined [here](https://discuss.elastic.co/t/kibana-unaccessible-after-using-migration-ui/101763). 

We do have an API available. The original PR is [here](https://github.com/elastic/kibana/pull/11632) with some details, and I am working now to get documentation up on our site.

**Comment by tylersmalley** [comment]
Alternatively, you could create an export which users could then import into their Kibana instance - though, they would need to manually add the index pattern prior as it's not currently part of the exported objects. This could be the quickest path to 6.0 for your project @felixbarny.

**Comment by tylersmalley** [comment]
There is also an import/export API for dashboards and their dependencies [here](https://github.com/elastic/kibana/pull/10858)

**Comment by daboom** [comment]
@felixbarny could you tell me where the core of the Elastic Search requests is done so I can try to contribute? Thank you.

**Comment by felixbarny** [comment]
Sorry for the late answer. Its ElasticsearchClient

**Comment by daboom** [comment]
Any news?

**Comment by daboom** [comment]
I tried what @tylersmalley suggested using the import/export api of Kibana. Here the steps:
1 - exported all dashboards from Kibana 5.6.5 using `api/kibana/dashboards/export`
2 - imported all dashboards into a clean Kibana 6.1.0 using `api/kibana/dashboards/import`
3 - started tomcat 8 with spring-petclinic-1.5.1 
4 - surfed around a bit on the app
5 - checked Kibana discover and dashboards

The documents are stored and the dashboards seem to work but I see a lot of errors on stage-monitor logs like:
```
2017-12-19 12:06:39.324  WARN 6192 --- [c-elasticsearch] Client$BulkErrorReportingResponseHandler : Error(s) while sending a _bulk request to elasticsearch:
 - Rejecting mapping update to [.kibana] as the final mapping would have more than 1 type: [doc, dashboard]
 - Rejecting mapping update to [.kibana] as the final mapping would have more than 1 type: [visualization, doc]
```
I see that all errors are related to .kibana write tentatives. Can I still use the dashboards as they are after the import or there is still something messed up?

Thank you

**Comment by felixbarny** [comment]
Closed by #365
